mesonbuild/modules/python.py

```diff
@@ -410,6 +416,19 @@ def _get_path(self, state: T.Optional['ModuleState'], key: str) -> None:
         value = state.get_option(f'{key}dir', module='python')
         if value:
             return value
+
+        install_env = state.get_option('install_env', module='python')
+        if install_env == 'auto':
+            install_env = 'venv' if self.info['is_venv'] else 'system'
+
+        if install_env == 'system':
+            rel_path = os.path.join(self.info['variables']['prefix'], rel_path)
+        elif install_env == 'venv':
+            if not self.info['is_venv']:
+                raise mesonlib.MesonException('python.install_env cannot be set to "venv" unless you are in a venv!')
+            # inside a venv, deb_system is *never* active hence info['paths'] may be wrong
+            rel_path = self.info['sysconfig_paths'][key]
+
         # Use python's path relative to prefix, and warn if that's not a location
         # python will lookup for modules.
         abs_path = Path(state.get_option('prefix'), rel_path)
```

Comment:
install_env cannot suppress the warning though, it merely gives meson a better hint about which heuristic path to choose.

Given the warning is really just about Debian's brokenness, it's not actually unlikely that Debian's bad patching results in actual sysconfig paths that aren't on the pythonpath.

So I believe it is inappropriate to update the warning below, unless setting `python.install_env` actually returns early, skipping the warning check altogether.

Reply:
The warning happens on Windows too, IIRC. Setting `python.install_env` should avoid the warning IMHO.

Reply:
Should, but isn't guaranteed...

I'm worried about printing "you can avoid this warning by setting install_env" and then people actually setting this and still getting warnings because they have a bizarre patched sysconfig, which we *think* is only the case on Debian.

Conditionally mentioning to use install_env depending on whether we guess that platform is bizarre, feels broken and implies we should conditionally change our detection logic instead, and maybe get rid of the warning. :p